src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -29,16 +29,29 @@
 
 /**
  * <p>
- * Checks that overload methods are grouped together.
+ * Checks that overloaded methods are grouped together. Overloaded methods have the same
+ * name but different signatures where the signature can differ by the number of
+ * input parameters or type of input parameters or both.
  * </p>
  * <p>
- * Example of incorrect grouping overload methods:
+ * Example of correct grouping of overloaded methods:
  * </p>
  * <pre>
  * public void foo(int i) {}
  * public void foo(String s) {}
- * public void notFoo() {} // Have to be after foo(int i, String s)
+ * public void foo(String s, int i) {}
  * public void foo(int i, String s) {}
+ * public void notFoo() {}
+ * </pre>
+ * <p>
+ * Example of incorrect grouping of overloaded methods:
+ * </p>
+ * <pre>
+ * public void foo(int i) {} //OK
+ * public void foo(String s) {} //OK
```

Comment:
sorry to nitpick, want to add a space here? `// OK`

Reply:
All nitpicks must be fixed, @esilkensen , please do not hesitate to raise them